scripts/prep-release.js

```diff
@@ -0,0 +1,150 @@
+import {
+	NOTICE_LEVEL,
+	abortAndSwitchToBranch,
+	checkBinaryExists,
+	createPullRequest,
+	getCurrentBranchName,
+	getCurrentVersion,
+	getNvmrcVersion,
+	gitFactory,
+	error,
+	info,
+	success,
+	isCorrectNodeVersion,
+	openPullRequest,
+	promptContinue,
+	switchToBranchWithMessage,
+} from './utils.js';
+
+import bumpVersion from './commands/bump-version.js';
+import updateReadMe from './commands/readme.js';
+import updateTranslations from './commands/update-translations.js';
+
+async function main() {
+	if ( ! isCorrectNodeVersion() ) {
+		return abortAndSwitchToBranch(
+			`Your version of NodeJS is not correct. Please install NodeJS v${ getNvmrcVersion() }.`,
+			NOTICE_LEVEL.ERROR
+		);
+	}
+
+	if ( ! checkBinaryExists( 'gh' ) ) {
+		error(
+			`The Github CLI (gh) is not installed. 
+			
+Please install it from https://cli.github.com/ or using 'brew install gh' if you're on a Mac.`
+		);
+		process.exit( 1 );
+	}
+
+	info( `This command will prepare a new wc-calypso-bridge release with the following steps:
+
+1. Bump the version number in the composer.json file and the wc-calypso-bridge.php file.
+2. Update the changelog in readme.txt.
+3. Update the translation files (optional).
+4. Create a new PR with the changes.
+	` );
+
+	if ( ! ( await promptContinue( 'Continue?' ) ) ) {
+		process.exit( 1 );
+	}
+
+	const currentBranchName = await getCurrentBranchName();
+
+	if ( ! ( await bumpVersion() ) ) {
+		error(
+			'Aborting, something went wrong while bumping the version number.'
+		);
+		await switchToBranchWithMessage( currentBranchName );
+		process.exit( 1 );
+	}
+
+	const changelogEntry = await updateReadMe();
+	if ( changelogEntry === false ) {
+		error( 'Aborting, something went wrong while updating the readme.' );
+		await switchToBranchWithMessage( currentBranchName );
+		process.exit( 1 );
+	}
+
+	if (
+		await promptContinue(
+			'Would you like to update the translation files?'
+		)
+	) {
+		await updateTranslations();
+	}
+
+	if (
+		! ( await promptContinue( 'Would you like to create a pull request?' ) )
+	) {
+		error( 'Pull request creationg was cancelled.' );
+		process.exit( 1 );
+	}
+
+	// Push our release prep branch
+	const git = gitFactory();
+	try {
+		await git.push( 'origin', git.branch(), [ '--set-upstream' ] );
```

Comment:
I think this maybe can solve the problem I am facing.
```suggestion
		await git.branch(['--set-upstream-to=origin/${branchName}', branchName]);
		await git.push( 'origin', git.branch() );
```